### DISCUSSION TITLE
What's wrong with resetting with both --mixed and paths?

### DISCUSSION THREAD

**Discussion by EliahKagan** [gitpython-developers/GitPython#1876] [Q&A] [answered]
The `HEAD.reset` method includes this code:

https://github.com/gitpython-developers/GitPython/blob/e880c33284f2abff55d3cddfef2302663b178005/git/refs/head.py#L97-L105

The issue URL is broken and seems it may not be archived. Is more known about this issue, and is there anything that can be replaced with?

**Comment by Byron** [comment]
Under the assumption that the project was moved from `Byron` to `GitPython-Developers`, I thought that https://github.com/GitPython-Developers/GitPython/issues/2 is the issue referred to here. However, it doesn't seem to be related at all.

Maybe one can try `--mixed` alongside some pathspecs on the command-line to see if it works or not with more recent Git versions?

**Comment by EliahKagan** [comment, marked as answer]
Although I can't be certain there isn't further relevant information in the lost issue, I think I've figured this out. It's much more straightforward than I'd feared and indeed trying it out as you suggested in https://github.com/gitpython-developers/GitPython/discussions/1876#discussioncomment-8784228 led directly to the answer.

Running <code>git reset --mixed -- *path*</code> always shows a warning. For example:

```text
> git diff --staged --stat
 git/__init__.py | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)
> git reset --mixed -- git/__init__.py
warning: --mixed with paths is deprecated; use 'git reset -- <paths>' instead.
Unstaged changes after reset:
M       git/__init__.py
```

The suggestion of what to do instead was added in https://github.com/git/git/commit/a4941a81c8a67abad2000b044ecad71ceff9d695, but the behavior of issuing a deprecation message is older than that.

It looks like it came in at https://github.com/git/git/commit/0e5a7faa3a903cf7a0a66c81e20a76b91f17faab when `git-reset` was changed from being implemented as a shell script `git-reset.sh` (retained as an example, then later removed in https://github.com/git/git/commit/49eb8d39c78f161231e63293df60f343d208f409 along with other such examples) to a builtin implemented in a newly introduced `builtin-reset.c`. The first stable version of `git` to include these changes was 1.5.4, whose [release notes](https://github.com/git/git/blob/master/Documentation/RelNotes/1.5.4.txt) mention `git reset` becoming a builtin but do not mention the deprecation of passing paths after `--mixed`. But it is not to be found in the old shell script.

As for why one *shouldn't* write `--mixed` when passing paths, it seems the idea is that a reset with paths--which must always have the same effect as when `--mixed` is passed--is already just its own separate kind of operation from other resets. Anyway, the approach taken in the code of simply omitting `--mixed`, not passing `--hard` or `--soft` either, and still passing the paths after `--`, seems to be robust and congruent with the expanded deprecation message.

I'll open a small PR to clarify the comment.
